test/nodl_to_policy/test_policy.py

```diff
@@ -0,0 +1,330 @@
+import nodl_to_policy.policy as policy
+import pytest
+import sros2
+
+
+def test_init_policy():
+    """Test that an initial empty policy is well-formed."""
+    test_policy = policy.init_policy()
+    assert test_policy.tag == 'policy'
+    assert test_policy.attrib['version'] == sros2.policy.POLICY_VERSION
+    assert len(test_policy) == 1
+    assert test_policy[0].tag == 'enclaves'
+    assert len(test_policy[0]) == 0
+
+
+def test_get_profile_minimal():
+    """Test an XML tree with no pre-existing `enclaves`/`profiles`/`profile` tags."""
+    test_policy = policy.init_policy()
+    test_profile = policy.get_profile(test_policy, node_name='foo')
+    # empty tags get generated after `get_profile` is called
+    test_enclaves = test_policy[0]
+    test_enclave = test_enclaves[0]
+    test_profiles = test_enclave[0]
+
+    # check that a single enclave tag was created
+    assert len(test_enclaves) == 1
+    assert test_enclave.tag == 'enclave'
+    assert test_enclave.attrib['path'] == '/'
+
+    # check that a single profiles tag was created
+    assert len(test_enclave) == 1
+    assert test_profiles.tag == 'profiles'
+
+    # check that a single profile tag was created
+    assert len(test_profiles) == 1
+    assert test_profiles[0] == test_profile
+    test_profile.tag == 'profile'
+    test_profile.attrib['ns'] == '/'
+    test_profile.attrib['node'] == 'foo'
+    assert len(test_profile) == 0
+
+
+def test_get_profile_exists(test_policy_tree):
+    """
+    Test an XML tree with existing `enclaves`/`profiles`/`profile` tags.
+
+    A lot of checks in this test are to ensure that the
+    `get_profile` function does not alter an existing policy tree.
+    """
+    test_policy = test_policy_tree
+    test_enclaves = test_policy.find(path='enclaves')
+    test_enclave = test_enclaves.find(path=f'enclave[@path="/"]')
+    test_profiles = test_enclave.find(path='profiles')
+    test_profile = policy.get_profile(test_policy, node_name='node_1')
+
+    # check that a single enclaves tag exists
+    assert test_policy[0] == test_enclaves
+    assert len(test_enclaves) == 1  # check that 'enclaves' tree contains one child ('enclave')
+    assert test_enclaves.tag == 'enclaves'
+
+    # check that a single enclave tag exists
+    assert test_enclaves[0] == test_enclave
+    assert len(test_enclave) == 1  # check that 'enclave' tree contains one child ('profiles')
+    assert test_enclave.tag == 'enclave'
+    assert test_enclave.attrib['path'] == '/'
+
+    # check that a single profiles tag exists
+    assert test_enclave[0] == test_profiles
+    assert len(test_profiles) == 2  # two nodes, one profile tag for each
+    assert test_profiles.tag == 'profiles'
+
+    # check that a <profile node='node1'> tag exists as expected
+    assert test_profiles.find(path=f'profile[@ns="/"][@node="node_1"]') == test_profile
+    test_profile.tag == 'profile'
+    test_profile.attrib['ns'] == '/'
+    test_profile.attrib['node'] == 'node_1'
```

Comment:
Same question about these dead statements; should they be `assert`ed?

Reply:
These should have been asserted as well. Fixed in b8a958c.